documentation/Makefile

```diff
@@ -8,20 +8,161 @@ EXPAND_ME += EPICS_MODIFICATION
 EXPAND_ME += EPICS_PATCH_LEVEL
 EXPAND_ME += OS_CLASS CMPLR_CLASS
 
-HTMLS += ComponentReference.html
-
-TARGETS += doxygen
-
 DOXYGEN = doxygen
 ME = documentation/O.$(T_A)/html
 GH_FILES = $(ME)/ $(ME)/.nojekyll $(ME)/*.* $(ME)/*/*.*
 
+c_TITLE = "Generic C/C++ API"
+
+c_HEADERS += adjustment
+c_HEADERS += alarm
+c_HEADERS += alarmString
```

Comment:
These lists need to be generated by a program. The existing Doxyfile just processes all of the files installed under `$(TOP)/include` (there are some headers in subdirectories of that which are relevant). If you don't want to include all of them, you should search for those that contain `@file` or `\file`, but I don't know how you would group them under different header types.

Reply:
I agree, but it is hard to get everything perfect in one go. This is a big and complicated PR, but at the same time difficult to split up in a meaningful way. I fully agree that we should not put in breaking changes or things that are difficult to revert. But this one seems to me like one that can be incrementally improved - unless I am missing something.
I would be happy to work on improvements as soon as we get this basic infrastructure in. Otherwise I start slowly to lose the hope that this will ever happen.

Reply:
+1 for gradual improvement

Reply:
I'm also in favor of leaving it for a later PR. I don't have a lot of time these days, so this would delay the PR further. Listing all header files is also in line with the Sphinx philosophy, IMHO, where we already have to list every document to include.